### DISCUSSION TITLE
User with role user can't logout

### DISCUSSION THREAD

**Discussion by IRediTOTO** [keystonejs/keystone#7519] [Questions] [answered]
Hi, I set role for user with access setting 
```
query: ({ session }) => session.data.role === 'admin',
```
I don't want normal user can see the user list
But user can't logout. Is there any solution for this now ? 
![image](https://user-images.githubusercontent.com/15695252/167286058-d4cbf3fa-3d61-4582-82fb-0fdef014e844.png)

**Comment by gwyneplaine** [comment, marked as answer]
Hey @IRediTOTO  we can't sign a user out if they don't have access to their own record. 
You probably want to augment your access control to look something like this.
```
filter: {
 query: ({ session }) => {
   if (!session) return false;
   if (session.role === 'admin') return true;
   return { id: { equals: session?.itemId } };
 }
}
```

This will ensure that: 
* the User list can't be queried if a user isn't signed in 
* All User records are available if you're an admin 
* You only have access to your own user record otherwise. 

Which should satisfy the use case you've provided

**Reply by IRediTOTO** [reply]
Thank you so much, hmm
